added: simple-timer @ 1.0.0

justjoeyuk opened the pull request:
# simple-timer

A minimal plugin designed for simplicity and ease of use.

- Recent Timers for quick access to common situations
- Subtle mode to be alerted in a less invasive fashion
- An alarm will sound and give you the option to suspend immediately or ignore

## Plugin Repo

- [x] I agree to and have already tested at least two plugins that are newly submitted or having a pending update to help ensure the health of the plugin testing process for all contributors.
- [x] [deck-roulette (1.1.6)](https://github.com/SteamDeckHomebrew/decky-plugin-database/pull/717)
- [x] [decky-game-settings](https://github.com/SteamDeckHomebrew/decky-plugin-database/pull/747)

## Checklist:

### Developer Checklist

- [x] I am the original author or an authorized maintainer of this plugin.
- [x] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [x] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [x] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.

### Plugin Backend Checklist

- **No**: I am using a custom backend other than Python.
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **No**: I am using a custom binary that has all of it's dependencies statically linked.

## Testing

- [x] Tested on SteamOS Stable/Beta Update Channel.

EMERALD0874 commented:
@justjoeyuk Please update your testing reports to conform to [this template](https://wiki.deckbrew.xyz/en/user-guide/testing). You need to give more information than what was provided.

EMERALD0874 commented:
@justjoeyuk The image in your plugins.json should be used to show your plugin or a graphic representing what it does, not advertise a website on the plugin store. decktools.xyz is fine in the author field, but please select a different image.

EMERALD0874 reviewed (dismissed):
@justjoeyuk The image in your plugins.json should be used to show your plugin or a graphic representing what it does, not advertise a website on the plugin store. decktools.xyz is fine in the author field, but please select a different image.

justjoeyuk commented:
@EMERALD0874 Understood. Promo image has been updated to: https://www.decktools.xyz/simple-timer-promo.jpeg and my testing reports have been updated to conform with the template.

EMERALD0874 commented:
Approved for deployment to testing store.

grimkor commented:
Was having a play with this in the morning and the timer appears to restart if you let the timer expire while the deck is in standby.

ohaiibuzzle commented:
I may be a bit picky on this, but:
- It appears that when the plugin will display `<0 minute` remaining near the timer's end, which might get a little confusing.
- Since you are only updating the timer in five-second intervals anyway (aka precision isn't a big concern), a way to implement this & avoid issues where the timer may get reset or stop counting (as a result of the Steam restarting or, as mentioned, going to sleep) would be to store the end time of the timer as Unix timestamp, and check against it instead of a sleeping loop. This would allow you to, eg. display a notification if the timer couldn't fire (due to the aforementioned reasons), or to implement multiple timers later on

justjoeyuk commented:
@ohaiibuzzle thanks for the suggestion, will handle implementation that way - good shout

EMERALD0874 commented:
@justjoeyuk Please do not check that someone has tested your plugin at the bottom of your PR template until a testing report has been submitted with the proper format. I've had to uncheck that for you twice.

Tormak9970 commented:
# Testing Report

### Versions
Decky: v3.1.2-pre1
SteamOS: v3.6.20 (Stable)

## Summary

### Major blocking issues
None

### Minor issues
None

### Currently installed plugins
 - **TabMaster** - Dev (essentially v2.8.0)
 - **CSS Loader** - 2.1.2
 - **SteamGridDB** - 1.4.0

### Yay/nay on to send to the production store or if it requires more testing.
Looks good to me

justjoeyuk commented:
I have updated the plugin to v1.0.0 with changes proposed for sleep management.

- Changed the button text to "Less than a minute" if seconds remaining < 60
- Timer is now based off a timestamp and will be compared when the plugin is either reloaded or device goes to sleep then wakes up
- If a timer has expired between plugin reloads or device wake, a simple toast will be shown notifying the user that the timer has expired and the timer will be removed
- If a timer has not expired between plugin reloads or device wake, the timer will continue until it was originally scheduled to end.

TrainDoctor commented:
@Tormak9970 just want you to confirm no blocking issues with the latest version.

EMERALD0874 reviewed (approved):
Tested this briefly on my Deck and had no blocking issues with the latest version. Also checked the latest features and they seem to work as intended. Checked the code and it also looks good. Considering this good to go!
